### ISSUE TITLE
Feature Request: Markdown editor instead of TinyMCE

### ISSUE THREAD

**Issue by jerry1970** [thorsten/phpMyFAQ#971]
I would love phpMyFAQ to have a Markdown editor instead of TinyMCE.

My team members tend to use different layouts and styles, etc. I don't want that.
Also, TinyMCE uses way too specific HTML styling which looks outdated and rather ugly.
It's a pain to change the way our FAQ articles look like when we decide to change the way the site looks.

I could remove most buttons from the TinyMCE toolbar and only let the editors select a style, but TinyMCE still adds too much HTML clutter.

Apparently I can switch off TinyMCE completely, but then I'd still need a Markdown parser in the front-end. I could probably add those in the templates myself, but it would be a great addition to phpMyFAQ when this would be just a switch in the configuration.

Also, a plain text editor would be OK, but having a Markdown preview tab (as on GitHub) would be wonderful.

So, in short, the feature request would be a configuration setting for Editor = TinyMCE (default), Markdown (plain text editor with Markdown help button and Markdown preview tab), plain.

**Comment by jerry1970** [comment]
Since we can't wait to have this implemented, could someone tell me which files to edit to (A) set a Markdown editor to the textarea where people edit a FAQ article and (B) let the article text be parsed by the Markdown parser? I assume it's only two files...

**Comment by thorsten** [comment]
Sure, here are the files:

* Disable the TinyMCE Editor in the admin configuration
* Add the Markdown Editor here: https://github.com/thorsten/phpMyFAQ/blob/2.8/phpmyfaq/admin/record.edit.php#L286
* The frontend parser needs to be added here: https://github.com/thorsten/phpMyFAQ/blob/2.8/phpmyfaq/artikel.php#L70

Would be cool if you could share your solution, I would add it to 2.9

**Comment by jerry1970** [comment]
Great, thanks for the quick reply! I have record.edit.php working, now to artikel.php.
We are using version 2.8.something right now, so I'll fork 2.9 later and share the solution.

I am thinking of a way to parse TinyMCE's HTML back to plain text or Markdown, so existing articles can be easily converted by edit + save.

**Comment by thorsten** [comment]
Awesome! Looking forward to your solution!

**Comment by jeanfabre** [comment]
That would be really nice to have this feature!

 I concur with the initial request, a markdown content will be a lot more standardized then free html editing, so the option would be very welcomed. It also will be very useful to serve content over the REST api, because other clients may not be html ready, and markdown will be easier to parse and will remain readable as is.

Jean

**Comment by jerry1970** [comment]
@thorsten, it's working but I don't have the time to get to know the proper phpMyFaq project structure to know where these files should be. I sent you an email with all the changes I made.

I am using ParsedownExtra with Prism for syntax highlighting. It's very easy to use different markdown parsers (I started with plain markdown). This could even be a setting in the PMF config.

**Comment by thorsten** [comment]
I will add your code in 2.9 with an option to enable/disable Markdown. Thanks!

**Comment by jerry1970** [comment]
Please take a good look at the code. For example, I didn't include a check for wysiwyg, plain, or markdown editor. If wysiwyg is still on, the markdown preview in the record.edit.php doesn't make a lot of sense, so a few extra if() blocks are probably required. Also a bit of cleaning up the code is required. Also, you could decide whether the Prism addons I selected are useful. Maybe you want all Prism addons optional, I leave that up to you. But at least all the code is there, it's working.

Now I have to think about converting TinyMCE articles into Markdown articles... I have a few hundred existing articles to go... ;)

**Comment by thorsten** [comment]
phpMyFAQ 2.9.0-alpha3 supports Markdown :-)

**Comment by jeanfabre** [comment]
Excellent, thanks for your hard work!

**Comment by jerry1970** [comment]
Great, looking forward to testing it! :+1:

**Comment by jerry1970** [comment]
@thorsten, where can I find 2.9.0 alpha 3?
Thanks!

**Comment by thorsten** [comment]
I'll release it as soon as possible!

**Comment by thorsten** [comment]
Should be fixed now.

**Comment by jerry1970** [comment]
Thanks, the setup now runs smoothly!

When running the index.php, the page remains blank. html has display:none;. Any ideas?

**Comment by thorsten** [comment]
Did you run 

$ npm install
$ bower install
$ grunt

**Comment by nighknight** [comment]
> Now I have to think about converting TinyMCE articles into Markdown articles... I have a few hundred existing articles to go... ;)
I am thinking of converting from HTML to Markdown as well. Did you complete your quest?

**Comment by thorsten** [comment]
@nighknight what do you mean?

**Comment by nighknight** [comment]
> @nighknight what do you mean?
I have a phpMyFAQ instance with ~ 1000 FAQs in three languages (DE/EN/FR, so 3000 FAQs in total) in HTML. New entries need to be translated in all three languages. 

Meanwhile, Deepl or Google Translate provide a very good starting point for these translations. However, the process of formatting the entry in the first language in TinyMCE and then doing a copy&paste for each paragraph to the translation tool is a tedious work (same the other way round). It is much easier if you use a markup language like Asciidoc or Markdown as input format, because here you can copy the whole text to the translation tool and keep the formatting. 

Hence, I am thinking of switching to Markdown for the FAQs, but this would require to convert all 3000 FAQs from HTML to Markdown. Hence the question if there is an easy way to do this... :-)
